### DISCUSSION TITLE
Aux heat control in a heat pump set up

### DISCUSSION THREAD

**Discussion by AlbertShovelton** [jmcollin78/versatile_thermostat#749] [Ideas]
I am using VTherm to control 2 heat pumps. With auto regulation on, vtherm does a decent job stabilizing the room temperature however on really cold days I need an auxiliary heater to help maintain temperature and assist with ramping. 

Currently, I control this with a heater on a switch, but since VTherm doesn’t know about the aux heater, it with turn down the heat pumps to compensate for the aux heater.

This might be out of scope for this project, but I’d be interested to see if there’s a way to integrate the aux heater into VTherm to automatically handle control based on some temperature threshold and a preconfigured temperature target where the heat pump looses capacity.

currently, I use an automation that monitors the difference between room temperature and target. When the difference is greater than 5 F, it turns the heater on. I can provide the yaml if it helps.

**Comment by jmcollin78** [comment]
Hello,

> since VTherm doesn’t know about the aux heater, it with turn down the heat pumps to compensate for the aux heater.

I don't understand this. There is no mecanism that "turn off to compensate for aux". VTherm `over_climate` I think it is your case, implements the aux API so it should work with aux heater. But I never try to test it (and I have to admit, I don't know how it is supposed to work)

**Comment by AlbertShovelton** [comment]
Ok, I went back and played with some settings on self regulation and it’s better now. I am still running into some equipment limitations. I.e. when the temperature is <20f the temperature still drops even when with a 5 degree offset. 

My system has boost function, which is able to bring the temperature back to target, but with the strong self regulation, vtherm sees a spike in temperature, so it drops the thermostat setpoint, creating a feed back loop where now the regulated set point is to low to maintain temperature and the room cools quickly, requiring another boost.

I want to try and fix this by using the medium regulation, but the temperature offset is low and I live in a cold weather climate. Would these modifications to the medium regulation preset below create a slower responding self regulation in a cold weather climate?

kp: 0.3
ki: 0.05
K_ext: 0.0
offset_max: 6
stabilization_threshold: 0.1
accumulated_error_threshold: 20

thanks!

**Comment by jmcollin78** [comment]
Hello @AlbertShovelton ,

Self-regulation is auto-adaptative (that is the main purpose). So, if there is a spike of temperature, you need less heating power and that is exactly what it does. Its only purpose it to keep the setpoint stable.

On the contrary if the temp drops, the regulation will augment the setpoint and then generate more heating power.

If you device is not enough powered for the cold weather, then turn on your auxiliary radiators, and Vtherm will automatically adapt the setpoint of the heatpump.

I have exactly the same type of configuration with wood stove and electrical radiator and a PAC. If I start the wood stove, Vtherms adapts the setpoint to the PAC and it is frequent that it turn it off because target temperature is exceeded. Idem for the electrical radiators that are here if the PAC is not sufficient (and wood stove off).

> Would these modifications to the medium regulation preset below create a slower responding self regulation in a cold weather climate?

Not sure of what you mean with 'slower responding'. Maybe lower the kp parameter will slower the self-regulation. You will have to try.

**Comment by AlbertShovelton** [comment]
Ok, did a little tinkering and finally had some cold weather to see how it works.  

First, I will need to use expert mode for regulation overnight. I copied and pasted the exact settings from the documents 'extreme regulation' into my config file, updated self-regulation for vtherm to expert and restarted my HA. The thermostat is regulating, correctly, but the offset doesn't go to the full offset, it seems to stop at +5 degrees where I think it should go to +10 degrees.

When thinking about my space heater, I'd always prefer to max out the heat pump, then turn on and cycle electric heater. However, this isn't how self-regulation works. With the space heater on, self-regulation will drop the underlying temperature not turn off the heater (obviously because vtherm doesn't know about the additional heater). My ideal situation would be:

1) room temperature drops below target.
2) self-regulation is at maximum offset
3) turn cycle aux heater to maintain temperature. 

I think I can do this by just monitoring the difference between regulated temperature and set-point, but it would be easier if there was a helper that signals that self-regulation is at it's maximum offset.
